packages/client/tests/functional/restart/tests.ts

```diff
@@ -0,0 +1,50 @@
+import { faker } from '@faker-js/faker'
+import { ClientEngineType, getClientEngineType } from '@prisma/sdk'
+import * as util from 'util'
+
+import testMatrix from './_matrix'
+
+const sleep = util.promisify(setTimeout)
+
+// @ts-ignore this is just for type checks
+declare let prisma: import('@prisma/client').PrismaClient
+// @ts-ignore this is just for type checks
+declare let PrismaClient: typeof import('@prisma/client').PrismaClient
+
+testMatrix.setupTestSuite(() => {
+  test('should assert that PrismaClient is restarted when it goes down', async () => {
+    // No child process for Node-API, so nothing that can be killed or tested
+    if (getClientEngineType() === ClientEngineType.Library) {
+      return
+    }
+
+    // Disconnect from the injected client - we only use this to apply migrations.
+    await prisma.$disconnect()
+
+    const client = new PrismaClient()
+    await client.$connect()
+
+    const username = faker.internet.userName()
+
+    await client.user.create({ data: { username } })
+
+    // @ts-ignore kill the binary child process
+    client._engine.child.kill()
+    await sleep(1000)
```

Comment:
Do we have any idea why we sleep here and in the loop below? Is it to make sure that process died? If so, maybe we should more explicitly wait for it: for example, use [this package](https://github.com/sindresorhus/process-exists)

Reply:
I could not use this because of jest and babel issues importing a module, however, I have simplified the logic and added an await process.exit function.